app/src/androidTest/java/com/example/tms07/CarTest.kt

```diff
@@ -0,0 +1,106 @@
+import com.example.tms07.database.MyDatabase
+import com.example.tms07.database.dao.CarDao
+import com.example.tms07.database.entity.Car
+import com.google.common.truth.Truth
+import kotlinx.coroutines.runBlocking
+import org.junit.Before
+import org.junit.Test
+
+class CarTest {
+
+    private lateinit var car: Car
+    private lateinit var carDao: CarDao
+
+    @Before
+    fun init() {
+        val context = InstrumentationRegistry.getInstrumentation().targetContext
+        val db = Room.inMemoryDatabaseBuilder(context, MyDatabase::class.java)
+            .allowMainThreadQueries()
+            .build()
+        carDao = db.carDao()
+    }
+
+    @Test
+    fun insert_returnTrue() {
+        car = Car("Audi", 2016, 250.000)
+
+        runBlocking {
+            carDao.insert(car)
+            val carList = carDao.getAll()
+            Truth.assertThat(carList.contains(car)).isTrue()
+        }
+    }
+
+    @Test
+    fun delete_returnFalse(){
+        car = Car("Audi", 2016, 250.000)
+        val carBmw = Car("BWM", 2010, 140.000)
+        val carMercedes = Car("Mercedes", 2005, 450.000)
+
+        runBlocking {
+        carDao.insert(car)
+        carDao.insert(carBmw)
+        carDao.insert(carMercedes)
+
+        carDao.deleteItem("BWM", 2010, 140.000)
+        val carList = carDao.getAll()
+        Truth.assertThat(carList.contains(carBmw)).isFalse()
+        }
+    }
+
+    @Test
+    fun searchBrand_returnFalse(){
+        car = Car("Audi", 2016, 250.000)
+        val carBmw = Car("BWM", 2010, 140.000)
+        val carAudi = Car("Audi", 2020, 500.000)
+        val carMercedes = Car("Mercedes", 2005, 450.000)
+
+        runBlocking {
+            carDao.insert(car)
+            carDao.insert(carBmw)
+            carDao.insert(carAudi)
+            carDao.insert(carMercedes)
+
+            val listOfCarsBySearch = carDao.searchBrand("Audi")
+           Truth.assertThat(listOfCarsBySearch.contains(carBmw)).isFalse()
+        }
+    }
+
+    @Test
+    fun searchBrand_returnTrue(){
+        car = Car("Audi", 2016, 250.000)
+        val carBmw = Car("BWM", 2010, 140.000)
+        val carAudi = Car("Audi", 2020, 500.000)
+        val carMercedes = Car("Mercedes", 2005, 450.000)
+
+        runBlocking {
+            carDao.insert(car)
+            carDao.insert(carBmw)
+            carDao.insert(carAudi)
+            carDao.insert(carMercedes)
+
+            val listOfCarsBySearch = carDao.searchBrand("Audi")
+            Truth.assertThat(listOfCarsBySearch.contains(carAudi)).isTrue()
+        }
+    }
+
+    @Test
+    fun getAll_returnTrue() {
+        val carBmw = Car("BWM", 2010, 140.000)
+        val carAudi = Car("Audi", 2020, 500.000)
+        val carMercedes = Car("Mercedes", 2005, 450.000)
+
+        runBlocking {
+            carDao.insert(carBmw)
+            carDao.insert(carAudi)
+            carDao.insert(carMercedes)
+
+            val carList = carDao.getAll()
+            Truth.assertThat(carList == carDao.getAll()).isTrue()
```

Comment:
Most likely it will be better here to compare the list of cars, which you white above, with the car list from carDao.getAll()

Reply:
I agree with the comment above.
It is equivalent to carDao.getAll() == carDao.getAll()

Reply:
So, what do you think how can I check whether the output of getAll () works correctly
Maybe it loses data when outputting?

Reply:
Create a list of your cars before adding them to the database. Then compare this list with the list from the database.

Reply:
Does it make sense to clog the memory with unnecessary arrays?)